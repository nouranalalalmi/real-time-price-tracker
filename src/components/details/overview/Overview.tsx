'use client';

import Image from 'next/image';
import { useParams } from 'next/navigation';

import { COINS } from '@/constants';
import { useGetAssetDetails } from '@/services/details';
import { FormattedNumber } from '@/ui/formattedNumber/FormattedNumber';
import { getColorsClasses } from '@/utils/numbers/getColorsClasses';

import { OverviewItem } from './OverviewItem';

export const Overview = () => {
  const { id } = useParams();
  const { data } = useGetAssetDetails(id as string);
  const logo = COINS.find(c => c.id === id)?.logo;

  return (
    <div className="flex w-full flex-col">
      <h1 className="pb-2 text-3xl font-semibold">Overview</h1>
      <div className="flex flex-col gap-y-4 rounded-lg border bg-white p-4 shadow-sm">
        <div className="flex items-center gap-2 border-b">
          <h2 className="text-2xl capitalize">{id}</h2>
          {logo && <Image src={`../${logo}`} width={25} height={25} alt="asset-logo" />}
        </div>
        <div className="flex flex-col gap-y-4 text-left">
          <OverviewItem label="Rank">#{data?.rank}</OverviewItem>
          <OverviewItem label="Symbol">{data?.symbol}</OverviewItem>
          <OverviewItem label="Market cap">
            <FormattedNumber value={data?.marketCapUsd} />
          </OverviewItem>
          <OverviewItem label="Volume change (24hr)">
            <FormattedNumber value={data?.volumeUsd24Hr} />
          </OverviewItem>
          <OverviewItem label="Price">
            <FormattedNumber prefix="$" value={data?.priceUsd} />
          </OverviewItem>
          <OverviewItem label="Price change (24hr)">
            <div className={getColorsClasses(data?.changePercent24Hr).text}>
              <FormattedNumber suffix="%" value={data?.changePercent24Hr} />
            </div>
          </OverviewItem>
          <p className="text-xs font-semibold text-yellow-600">*Refetched every 3 minutes</p>
        </div>
      </div>
    </div>
  );
};
