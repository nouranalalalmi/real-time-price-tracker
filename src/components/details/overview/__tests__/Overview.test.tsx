import { render } from '@testing-library/react';
import mockRouter from 'next-router-mock';

import { Overview } from '../Overview';

jest.mock('@/services/details');

describe('Overview', () => {
  beforeEach(() => {
    mockRouter.push({ pathname: '/:id', query: { id: 'bitcoin' } });
  });

  it('should render the asset details when data is loaded', () => {
    const { container } = render(<Overview />);

    expect(container).toMatchSnapshot();
  });
});
