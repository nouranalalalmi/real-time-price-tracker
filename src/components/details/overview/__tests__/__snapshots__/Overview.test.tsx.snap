// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should render the asset details when data is loaded 1`] = `
<div>
  <div
    class="flex w-full flex-col"
  >
    <h1
      class="pb-2 text-3xl font-semibold"
    >
      Overview
    </h1>
    <div
      class="flex flex-col gap-y-4 rounded-lg border bg-white p-4 shadow-sm"
    >
      <div
        class="flex items-center gap-2 border-b"
      >
        <h2
          class="text-2xl capitalize"
        >
          bitcoin
        </h2>
        <img
          alt="asset-logo"
          data-nimg="1"
          decoding="async"
          height="25"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fbtc.png&w=64&q=75"
          srcset="/_next/image?url=%2Fimages%2Fbtc.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Fbtc.png&w=64&q=75 2x"
          style="color: transparent;"
          width="25"
        />
      </div>
      <div
        class="flex flex-col gap-y-4 text-left"
      >
        <div
          class="flex flex-col gap-y-1 break-all text-xl text-gray-700"
        >
          <h2
            class="text-sm font-bold"
          >
            Rank
          </h2>
          #
          1
        </div>
        <div
          class="flex flex-col gap-y-1 break-all text-xl text-gray-700"
        >
          <h2
            class="text-sm font-bold"
          >
            Symbol
          </h2>
          BTC
        </div>
        <div
          class="flex flex-col gap-y-1 break-all text-xl text-gray-700"
        >
          <h2
            class="text-sm font-bold"
          >
            Market cap
          </h2>
          1,154,471,549,507.90
        </div>
        <div
          class="flex flex-col gap-y-1 break-all text-xl text-gray-700"
        >
          <h2
            class="text-sm font-bold"
          >
            Volume change (24hr)
          </h2>
          13,866,421,449.48
        </div>
        <div
          class="flex flex-col gap-y-1 break-all text-xl text-gray-700"
        >
          <h2
            class="text-sm font-bold"
          >
            Price
          </h2>
          $
          58,623.43
        </div>
        <div
          class="flex flex-col gap-y-1 break-all text-xl text-gray-700"
        >
          <h2
            class="text-sm font-bold"
          >
            Price change (24hr)
          </h2>
          <div
            class="text-green-500"
          >
            2.24
            %
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
